onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlDFT.h

```diff
@@ -232,6 +230,15 @@ class GpuDFTOperator : public WRL::Base<IMLOperatorKernel>
         ORT_THROW_IF_FAILED(context->GetInputEdgeDescription(0, &edgeDesc));
         assert(edgeDesc.edgeType == MLOperatorEdgeType::Tensor);
 
+        if (version == 17)
+        {
+            ORT_THROW_IF_FAILED(context->GetAttribute("axis", MLOperatorAttributeType::Int, 1, sizeof(int64_t), reinterpret_cast<void*>(&m_axis)));
+        }
+        else
+        {
+            m_axis = -2;
```

Comment:
can you add a comment explaining why it's -2? why is that the default for non version 17? Also, how does non-opset 17 specificy the axis? 

and is the version check supposed to be just for 17 vs not-17, or is it intended to be 17 vs >17? What if the version is < 17?

Reply:
According to the documentation (https://github.com/onnx/onnx/blob/main/docs/Changelog.md#DFT-20) the default axis value for DFT-20 is -2 (the last signal axis). The check is only 17 vs not-17 since the onnx operator only has versions 17 and 20. Will add a comment explaining this.

Reply:
I think my point is about technical correctness vs practicality. Even though in practice the "else" block won't get hit when the opset is < 17 because that's not valid, this particular method still allows it, and if someone called it incorrectly it would execute that block. The technically correct check is that the else block should only execute when the version > 17. It's also worth thinking about what the design would be for a new opset: say opset 25 comes out and you want to add a new check here - how would that look?